modules/data.land/R/fertilizer_components.R

```diff
@@ -0,0 +1,110 @@
+#' Calculate the Nitrogen and Carbon Content of a Fertilizer Application
+#'
+#' This function calculates the different forms of nitrogen (NO3-N, NH4-N, organic N) and organic carbon (C_org) in a fertilizer application.
+#' It can determine fertilizer nitrogen and carbon content using either a lookup table based on 
+#' the SWAT model's [`fertilizer.frt`](https://github.com/swat-model/swatplus/blob/main/data/Osu_1hru/fertilizer.frt)
+#' file or determine the fertilizer's 
+#' nutrient content based on NN-PP-KK format.
+#'
+#' Consistent with assumptions (TODO: confirm this statement) in DayCent, DSSAT, and other models, urea is treated as NH3 because the transformation typically occurs within a day.
+#' 
+#' @param fertilizer_type Character string specifying the type of fertilizer. Can be one of "urea", "ammonium_nitrate", "compost", "manure", etc., or a string in the format "NN-PP-KK" (e.g., "45-5-10").
+#' @param fertilizer_amount Numeric value specifying the amount of fertilizer applied in kg/ha.
+#' @param fraction_organic_n Optional numeric value specifying the fraction of nitrogen in the fertilizer. Used for organic fertilizers if not provided in the dataset.
+#' @param fraction_organic_c Optional numeric value specifying the fraction of carbon in the fertilizer. Used for organic fertilizers if not provided in the dataset.
+#'
+#' @return A list containing:
+#'   - `fertilizer_type`: The type of fertilizer used.
+#'   - `NO3_N`: The amount of nitrate nitrogen (NO3-N) in kg/ha.
+#'   - `NH4_N`: The amount of ammonium nitrogen (NH4-N) in kg/ha.
+#'   - `N_org`: The amount of organic nitrogen in kg/ha.
+#'   - `C_org`: The amount of organic carbon in kg/ha.
+#'
+#' @examples
+#' fertilizer_components("45-00-00", 200)
+#' fertilizer_components("compost", 1000)
+#' fertilizer_components("manure", 1000, fraction_organic_n = 0.02, fraction_organic_c = 0.08)
```

Comment:
I don't see either "compost" or "manure" as names in the SWAT table -- will these examples actually work as written?